src/Lykke.Tool.BitcoinCashAddressTransformer/Program.cs

```diff
@@ -0,0 +1,207 @@
+                {
+                    throw new ArgumentException($"Unable to recognize address {observableWallet.Address}", nameof(observableWallet.Address));
+                }
+
+                var oldAdrr = addr.ScriptPubKey.GetDestinationAddress(network).ToString();
+                var newAddr = addr.ScriptPubKey.GetDestinationAddress(bcashNetwork).ToString();
+
+                if (!string.Equals(observableWallet.Address, oldAdrr, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    var prevColor = Console.ForegroundColor;
+                    Console.ForegroundColor = ConsoleColor.Yellow;
+                    Console.WriteLine($"{observableWallet.Address} not in obsolete format. Old format: {oldAdrr} | New Format {newAddr}");
+                    Console.ForegroundColor = prevColor;
+                }
+
+                return newAddr;
+            });
+
+            var backupProgressCounter = 0;
+            Console.WriteLine($"Starting {observableWallets.Count} observable wallets backup to {backupTableName} table");
+
+            foreach (var observableWallet in observableWallets)
+            {
+                backupProgressCounter++;
+                Console.WriteLine($"Backup {observableWallet.Address} -- {backupProgressCounter} of {observableWallets.Count}");
+
+                await observableWalletBackupRepository.Insert(observableWallet);
+            }
+
+            var flushingProgress = 0;
+            Console.WriteLine("Flushing balance table");
+
+            foreach (var observableWallet in observableWallets)
+            {
+                flushingProgress++;
+                Console.WriteLine($"Deleting wallet balance record {observableWallet.Address} -- {flushingProgress} of {observableWallets.Count}");
+
+                await walletBalanceRepo.DeleteIfExist(observableWallet.Address);
+            }
+
+
+            var removingProgress = 0;
+            Console.WriteLine("Clearing table ObservableWallets");
+
+            foreach (var observableWallet in observableWallets)
+            {
+                removingProgress++;
+                Console.WriteLine($"Deleting observable wallet record -- {removingProgress} of {observableWallets.Count}");
+
+                await observableWalletRepository.Delete(observableWallet.Address);
+            }
+
+            var refillingProgress = 0;
+            Console.WriteLine("Refilling table ObservableWallets");
+
+            foreach (var observableWallet in observableWallets)
+            {
+                refillingProgress++;
+                var newAddress = obserwabletWalletsTransformation[observableWallet.Address];
+
+                Console.WriteLine($"Inserting obserwablewallet record {observableWallet.Address} => {newAddress} " +
+                                  $"-- {refillingProgress} of {observableWallets.Count}");
+
+                await observableWalletRepository.Insert(ObservableWallet.Create(newAddress));
```

Comment:
Let's group wallets by PK and do batch insert + in parallel thread limited by 8.